bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,329 @@
+            self.messages[message.id] = {  # store it
+                'word': word,  # store the word
+                'author': message.author,  # and the author
+                'score': 0,  # and the score
+            }
+
+            for emoji in self.emojis_val.keys():  # get the emojis
+                await message.add_reaction(emoji)  # and add them
+
+            logger.info(f"{message.author} added the word {word!r}")
+            # display if entry was successful
+            return await ctx.send(f"{word!r} added successfully!\n{random.choice(self.added_messages)}")
+
+        else:
+
+            logger.info(f"{ctx.message.author} tried to add a word, but the poll had already started.")
+            await ctx.send("Sorry, the poll has started! You can try and participate in the next round though!")
+
+    @commands.Cog.listener()
+    async def on_reaction_add(self, reaction: discord.Reaction, user: discord.User) -> None:
+        """Ensures that each user adds one and only one reaction."""
+        try:
+            if reaction.emoji in self.emojis_val.keys() and reaction.message.id in self.messages.keys():
+                msg = reaction.message
+                msg: Message
+                # create a custom counter
+                counter = {}
+                for r in msg.reactions:
+                    async for user in r.users():
+                        if user in counter:
+                            counter[user] += 1
+                        else:
+                            counter[user] = 1
+
+                if counter[user] > 1 and user != self.bot.user:  # If the user has more than one reaction
+                    await user.send("Sorry, you have already added a reaction, please remove your reaction and try \
+again")
+                    return await reaction.remove(user)  # removes user's reaction
+        except RuntimeError:  # The dictionary was changed in between
+            pass
+
+    @spook_name_rate.command(name='delete')
+    async def delete_word(self, ctx: Context) -> None:
+        """Delete's the user's word."""
+        if not self.poll:
+            for message_id in self.messages.keys():
+                data = self.messages[message_id]
+                if ctx.author == data['author']:
+                    del self.messages[message_id]  # delete the data
+                    if self.messages.get(message_id) is None:  # Make sure it does not exist
+                        # display if deletion successful
+                        return await ctx.send(f'Message deleted successfully ({data["word"]!r})!')
+                    else:
+                        # display if deletion unsuccessful
+                        return await ctx.send("Oops, there was some error, please try again!")
+            return ctx.send("But you don't have an entry... :eyes: Type `spooknamerate `")
+        else:
+            await ctx.send("You can't delete your word since the poll has already started!")
```

Comment:
Fail fast instead of wrapping whole logic in if statement

Reply:
okie done 😄